satellite/console/consoleweb/server.go

```diff
@@ -106,6 +112,116 @@ func (s *Server) appHandler(w http.ResponseWriter, req *http.Request) {
+
+	RemoteStoredData string
+	InlineStoredData string
+	RemoteSegments   string
+	InlineSegments   string
+	Objects          string
+	MetadataSize     string
+
+	RepairEgress string
+	GetEgress    string
+	AuditEgress  string
+
+	Since  string
+	Before string
+}
+
+// bucketUsageReportHandler generate bucket usage report page for project
+func (s *Server) bucketUsageReportHandler(w http.ResponseWriter, req *http.Request) {
+	var err error
+
+	var projectID *uuid.UUID
+	var since, before time.Time
+
+	tokenCookie, err := req.Cookie("tokenKey")
+	if err != nil {
+		s.log.Error("bucket usage report error", zap.Error(err))
+
+		w.WriteHeader(401)
+		http.ServeFile(w, req, filepath.Join(s.config.StaticDir, "static", "errors", "404.html"))
+		return
+	}
+
+	auth, err := s.service.Authorize(auth.WithAPIKey(context.Background(), []byte(tokenCookie.Value)))
+	if err != nil {
+		w.WriteHeader(401)
+		http.ServeFile(w, req, filepath.Join(s.config.StaticDir, "static", "errors", "404.html"))
+		return
+	}
+
+	defer func() {
+		if err != nil {
+			w.WriteHeader(404)
+			http.ServeFile(w, req, filepath.Join(s.config.StaticDir, "static", "errors", "404.html"))
+		}
+	}()
+
+	// parse query params
+	projectID, err = uuid.Parse(req.URL.Query().Get("projectID"))
+	if err != nil {
+		return
+	}
+	since, err = time.Parse(time.RFC3339, req.URL.Query().Get("since"))
+	if err != nil {
+		return
+	}
+	before, err = time.Parse(time.RFC3339, req.URL.Query().Get("before"))
+	if err != nil {
+		return
+	}
+
+	s.log.Debug("querying bucket usage report",
+		zap.String("projectID", projectID.String()),
+		zap.String("since", since.String()),
+		zap.String("before", before.String()))
+
+	ctx := console.WithAuth(context.Background(), auth)
+	bucketRollups, err := s.service.GetBucketsUsageRollups(ctx, *projectID, since, before)
+	if err != nil {
+		return
+	}
+
+	report, err := template.ParseFiles(path.Join(s.config.StaticDir, "static", "reports", "UsageReport.html"))
+	if err != nil {
+		return
+	}
+
+	precise := func(n float64) string {
+		return fmt.Sprintf("%.6f", n)
+	}
+
+	var displayRollups []displayBucketRollup
+	for _, bucketRollup := range bucketRollups {
+		displayRollup := displayBucketRollup{
+			BucketName: string(bucketRollup.BucketName),
+
+			RemoteStoredData: precise(bucketRollup.RemoteStoredData),
+			InlineStoredData: precise(bucketRollup.InlineStoredData),
+			RemoteSegments:   precise(bucketRollup.RemoteSegments),
+			InlineSegments:   precise(bucketRollup.InlineSegments),
+			Objects:          precise(bucketRollup.Objects),
+			MetadataSize:     precise(bucketRollup.MetadataSize),
+
+			RepairEgress: precise(bucketRollup.RepairEgress),
+			GetEgress:    precise(bucketRollup.GetEgress),
+			AuditEgress:  precise(bucketRollup.AuditEgress),
+
+			Since:  bucketRollup.Since.Format(time.RFC822),
+			Before: bucketRollup.Before.Format(time.RFC822),
+		}
+
+		displayRollups = append(displayRollups, displayRollup)
+	}
+
+	err = report.Execute(w, displayRollups)
```

Comment:
Shouldn't we do something with this error?

Reply:
each error is handled by defer function

`	defer func() {
		if err != nil {
			w.WriteHeader(404)
			http.ServeFile(w, req, filepath.Join(s.config.StaticDir, "static", "errors", "404.html"))
		}
	}()`

Reply:
ok, thanks